qiskit/visualization/pulse/matplotlib.py

```diff
@@ -504,8 +507,15 @@ def _draw_channels(self, ax, output_channels, interp_method, t0, tf, dt, v_max,
                     re, im = np.zeros_like(time), np.zeros_like(time)
                 color = self._get_channel_color(channel)
                 # scaling and offset
-                re = v_max * re + y0
-                im = v_max * im + y0
+                test1 = bool(min(re[~np.isnan(re)]) >= 0 and min(im[~np.isnan(re)]) >= 0)
```

Comment:
This seems to change the scaling of a channel with no negative waveform, but independently changing the waveform scaling might be confusing because we don't have pulse amplitude notation on vertical axis. I recommend to change y0 offset to remove redundant spacing when the waveform has no negative value.

https://github.com/Qiskit/qiskit-terra/blob/a934b1f948854b754c0e9518f307ab3d0f6daa6b/qiskit/visualization/pulse/matplotlib.py#L538

And removing redundant spacing should be style sheet option with default=`True`.

https://github.com/Qiskit/qiskit-terra/blob/a934b1f948854b754c0e9518f307ab3d0f6daa6b/qiskit/visualization/pulse/qcstyle.py#L18